stock_scrap_location_default/models/stock_scrap.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+from odoo.addons.stock.models.stock_location import Location as StockLocation
+
+
+class StockScrap(models.Model):
+
+    _inherit = "stock.scrap"
+
+    def _get_default_scrap_location_id(self) -> StockLocation:
+        """
+        Returns the default scrap location defined on company level
+        """
+        if self.env.company.scrap_default_location_id:
```

Comment:
This is not multi-company friendly as you are not using the company of the record

Reply:
The record at this step is being created (not yet). So, the scrap will be created with the company_id of the Environment.

Reply:
@jbaudoux ROADMAP added